gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp

```diff
@@ -1554,11 +1554,9 @@ static OGRFeature* OGR2SQLITE_FeatureFromArgs(OGRLayer* poLayer,
                     case OFTDate:
                     case OFTTime:
                     case OFTDateTime:
-                    {
-                        if( !OGRSQLITEStringToDateTimeField( poFeature, i, pszValue ) )
+                        if( !OGRParseDate( pszValue, poFeature->GetRawFieldRef(i), 0 ) )
```

Comment:
can you call poFeature->UnsetField(i) when OGRParseDate() fails ? Similarly for the other call sites of OGRSQLITEStringToDateTimeField(). Or maybe keep OGRSQLITEStringToDateTimeField() to do just that: call OGRParseDate() and UnsetField() when it fails ?